### PR TITLE
mock-pop-mining: implement VeriBlockPopMiner; add tests

### PR DESCRIPTION
**Pull request by eu321**

### CODE REVIEW

File: mock-pop-mining/src/main/java/org/veriblock/sdk/mock/VeriBlockPopMiner.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                tx.getBitcoinTransaction(),
+                tx.getMerklePath(),
+                tx.getBlockOfProof(),
+                tx.getBlockOfProofContext(),
+                signature,
+                tx.getPublicKey(),
+                tx.getNetworkByte());
+    }
+
+    public VeriBlockPublication mine(VeriBlockBlock blockToEndorse, VeriBlockBlock lastKnownVBKBlock, BitcoinBlock lastKnownBTCBlock, KeyPair key) throws SQLException, SignatureException, InvalidKeyException, NoSuchAlgorithmException {
```
Comment (Warchant) [inline mock-pop-mining/src/main/java/org/veriblock/sdk/mock/VeriBlockPopMiner.java:123]: Maybe just init this class with KeyPair and store it there? to not pass keypair on every call?

File: mock-pop-mining/src/test/java/org/veriblock/sdk/mock/VeriBlockBlockDataTest.java [hunk 2]
```diff
@@ -57,19 +73,117 @@ private void checkBlockData(VeriBlockBlockData blockData) {
...
+        byte[] data = new byte[count];
+        random.nextBytes(data);
+        return data;
+    }
+
+    private Sha256Hash randomSha256Hash(Random random) {
+        return Sha256Hash.wrap(randomBytes(random, Sha256Hash.BITCOIN_LENGTH));
+    }
+
+    private Address randomAddress(Random random) {
```
Comment (Warchant) [inline mock-pop-mining/src/test/java/org/veriblock/sdk/mock/VeriBlockBlockDataTest.java]: this looks like a good utility function, maybe you can create a class, which is created with `Random` instance?

Comment (eu321) [reply mock-pop-mining/src/test/java/org/veriblock/sdk/mock/VeriBlockBlockDataTest.java]: this is a good idea. probably can do it as part of the test refactoring task that is in the todo

File: mock-pop-mining/src/main/java/org/veriblock/sdk/mock/VeriBlockPopMiner.java (hunk 1 above)
Comment (eu321) [reply mock-pop-mining/src/main/java/org/veriblock/sdk/mock/VeriBlockPopMiner.java:123]: It is not yet clear as to what the final API would look like. For an attack simulation use case, ideally we want all inputs to mine() to be configurable eg several agents mining who knows what. In practice, I suspect I'll have to split this function into smaller ones as the first step and then address use cases one by one.

**Review by Warchant — APPROVED** [review]
Pls rebase on develop before merge.
Also do not forget to extract `random*` into a separate utility class

**Pull request merged**